spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RuntimeStreamsController.java

```diff
@@ -37,12 +37,17 @@
 import org.springframework.web.bind.annotation.RestController;
 
 /**
+ * @deprecated
+ * Deprecated and replaced by the {@link RuntimeStreamsControllerV2} instead.
+ * The new controller uses the /runtime/streams/{streamNames} query syntax.
+ *
  * Exposes runtime status of deployed streams.
  *
  * @author Christian Tzolov
  */
 @RestController
 @RequestMapping("/runtime/streams")
+@Deprecated
```

Comment:
Instead of deprecating this class, can we move the new changes into this controller itself and have the new REST endpoint updated?

Reply:
Currently the old (non compliant) REST endpoints are used by the UI for the metrics visualization (for example the dots under the deployed apps).  This as temp hack back then to transit from the pre-micrometer metrics collection.  
In short, imo we should keep the old endpoint until we migrate the UI to use the new one. And then we can remove it. 
The UI work could involving as we may have to change/adjust the JS domain model away from the proprietary classes used in the old endpoint.